lib/iot.dart

```diff
@@ -0,0 +1,583 @@
+                        fontSize: 16.0,
+                        color: Colors.grey,
+                      ),
+                    ),
+                    Container(
+                      margin: EdgeInsets.only(bottom: 40.0, top: 40.0),
+                      child: Center(
+                        child: ElevatedButton.icon(
+                          onPressed: () async {
+                            _pushConnectionRoute(context, scanCallback, connectionInfo);
+                          },
+                          style: ElevatedButton.styleFrom(
+                              primary: Theme.of(context).primaryColor.withOpacity(0.5),
+                              shape: RoundedRectangleBorder(
+                                borderRadius: BorderRadius.circular(50.0),
+                              ),
+                              padding: EdgeInsets.symmetric(horizontal: 20.0, vertical: 10.0)),
+                          label: Text(
+                            'Bæta við tengingu',
+                            style: TextStyle(color: Colors.white),
+                          ),
+                          icon: Icon(
+                            Icons.add,
+                            color: Colors.white,
+                          ),
+                        ),
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+          ),
+          Visibility(
+            visible: !isNetworkConnection,
+            child: Center(
+              child: Container(
+                margin: const EdgeInsets.only(
+                  top: 20.0,
+                  left: 25.0,
+                  bottom: 30.0,
+                  right: 25.0,
+                ),
+                child: Text(
+                  kNoInternetMessage,
+                  style: TextStyle(
+                    fontSize: 16.0,
+                    color: Colors.grey,
+                  ),
+                ),
+              ),
+            ),
+          ),
+          Visibility(
+            visible: isServerError,
+            child: Center(
+              child: Container(
+                margin: const EdgeInsets.only(
+                  top: 20.0,
+                  left: 25.0,
+                  bottom: 30.0,
+                  right: 25.0,
+                ),
+                child: Text(
+                  kServerErrorMessage,
+                  style: TextStyle(
+                    fontSize: 16.0,
+                    color: Colors.grey,
+                  ),
+                ),
+              ),
+            ),
+          ),
+          const SizedBox(
+            height: 50.0,
+          ),
+          Center(
+            child: Visibility(
+              visible: isSearching && !isServerError,
+              child: SpinKitRing(
+                color: Theme.of(context).primaryColor,
+                size: 50.0,
+              ),
+            ),
+          ),
+        ],
+      ),
+    ),
+  ];
+}
```

Comment:
OK, that is an incomprehensible monster. Do people generally do stuff like this in Flutter?